packages/codemirror-extension/src/index.ts

```diff
@@ -233,13 +277,70 @@ function activateEditorCommands(
     }
   });
 
+  /**
+   * Handle the settings of new widgets.
+   */
+  notebookTracker.widgetAdded.connect((sender, widget) => {
+    widget.content.widgets.forEach(cell => {
+      if (cell.inputArea.editor instanceof CodeMirrorEditor) {
+        const cm = cell.inputArea.editor.editor;
+        // Do not set scrollPastEnd option.
+        cm.setOption('keyMap', keyMap);
+        cm.setOption('theme', theme);
+        cm.setOption('styleActiveLine', styleActiveLine);
+        cm.setOption('styleSelectedText', styleSelectedText);
+        cm.setOption('selectionPointer', selectionPointer);
+        cm.setOption('lineWiseCopyCut', lineWiseCopyCut);
+      }
+    });
+  });
+
+  notebookTracker.activeCellChanged.connect(onActiveCellChanged, this);
+
+  let activeCell: Cell | null = null;
+
+  commands.addCommand('codemirror:leave-vim-insert-mode', {
+    label: 'Leave VIM Insert Mode',
+    execute: args => {
+      if (activeCell) {
+        let editor = activeCell.editor as CodeMirrorEditor;
```

Comment:
I think this is implicitly depending on `onActiveCellChanged` having been called in order to have set `activeCell`. 
```suggestion

  commands.addCommand('codemirror:leave-vim-insert-mode', {
    label: 'Leave VIM Insert Mode',
    execute: args => {
      const activeCell = notebookTracker.activeCell;
      if (activeCell) {
        let editor = activeCell.editor as CodeMirrorEditor;
```